### ISSUE TITLE
LT 5.9 causes extreme slowdown of context-menu in LibreOffice

### ISSUE THREAD

**Issue by ColeBantam** [languagetool-org/languagetool#7420]
Language Tool 5.9 makes the context-menu (right click) extremely slow in LibreOffice (7.3.6, 64Bit, Windows). With LT5.7 the context-menu opens without any noticeable delay, but with 5.9 it takes very long, sometimes seconds, till the context-menu appears. That makes it almost impossible to work with 5.9 so I had to revert to 5.7.

I have no copy of the LT 5.8 LibO-Plugin, so I can't test if that regression already exists in this version.

**Comment by Tex2002ans** [comment]
> I have no copy of the LT 5.8 LibO-Plugin, so I can't test if that regression already exists in this version.

You can grab the OXT files from this link:

- https://languagetool.org/download/

- - -

I don't have a delay on Right-Click in:

	LanguageTool 5.9 (2022-09-28 14:38:38 +0000, f8931be)
	OS: Windows 10 10.0 (amd64)
	LibreOffice 7.4.2.3 (The Document Foundation), en-US
	Java version: 17 (Oracle Corporation)
	Java max/total/free memory: 4082MB, 348MB, 173MB

and:

	Version: 7.4.2.3 (x64) / LibreOffice Community
	Build ID: 382eef1f22670f7f4118c8c2dd222ec7ad009daf
	CPU threads: 8; OS: Windows 10.0 Build 19045; UI render: Skia/Raster; VCL: win
	Locale: en-US (en_US); UI: en-US
	Calc: threaded

- - -

**Comment by ColeBantam** [comment]
I have retestet the issue with LanguageTool 5.8, and the issue does not exist in 5.9. So the regression was introduced between 5.8 and 5.9.

> I don't have a delay on Right-Click in:
I will add an test-file which one can use to reproduce the issue. For example: Open the text-file and go to page 90 and rightclick the link "'Sicherheitslücken in Firefox, Thunderbird und Tor Browser geschlossen' auf Heise Online". With languagetool 5.9 you should notice a big delay till the context-menu is displayed. Do the same with the following links to reproduce the problem. Clicking the same link twice does not trigger the effect!

**Comment by FredKruse** [comment]
@ColeBantam Did you test the Version 5.9.1? You could download it here: 
https://extensions.libreoffice.org/assets/downloads/99/1667923733/LanguageTool-5.9.1.oxt

**Comment by ColeBantam** [comment]
> @ColeBantam Did you test the Version 5.9.1? You could download it here: https://extensions.libreoffice.org/assets/downloads/99/1667923733/LanguageTool-5.9.1.oxt

Hi Fred.
I can not see any improvement in 5.9.1 regarding context-menu-speed.
Rolling back to 5.8 again...

**Comment by ColeBantam** [comment]
Tested Version 6.0 -> still unusable due to massive lag when right-clicking for context-menu. Context-menu can take up to several seconds if LT6 is installed. Rolling back to 5.8 again.

**Comment by nschipperbrainsmith** [comment]
Experiencing the same problem, running version 6 UI is so sluggish that it breaks libre office repeatedly.  Trying to use version 5.8 results in the error that automatic grammer check is disabled. So couldn't test using that version.

**Comment by Tex2002ans** [comment]
> Experiencing the same problem, running version 6 UI is so sluggish that it breaks libre office repeatedly.

What is your:

1. **Help > About LibreOffice** info?
2. **Tools > LanguageTool > About** info?

Is it happening in a specific document? Or all documents?

**Comment by nschipperbrainsmith** [comment]
1)  LibreOffice info
```
Version: 7.5.4.2 (X86_64) / LibreOffice Community
Build ID: 50(Build:2)
CPU threads: 16; OS: Linux 6.2; UI render: default; VCL: gtk3
Locale: nl-NL (en_US.UTF-8); UI: en-US
Ubuntu package version: 4:7.5.4~rc2-0ubuntu0.22.04.1~lo1
Calc: threaded
```

Unable to get the LanguageTool about info. But it was the latest stable (https://languagetool.org/download/) I tried based on this thread. From the download page. It right now complains about: "It seems the automatic grammar check is disabled. Please enable "Check grammar as you type" at ... " While this setting is enabled. 

An important note here which might help, I previously only used the new build in support for LanguageTool remote grammar checking. And not the extension directly. I'm guessing that makes this a bug report for LibreOffice rather than you, maybe?

**Comment by ColeBantam** [comment]
OK, this is an already known issue:

https://github.com/languagetool-org/languagetool/issues/8768

LibreOffice 7.5.4 breaks down the  LT-Plugin :( They hope it will be fixed in 7.5.5, but it doesn't look good currently :/

**Comment by ColeBantam** [comment]
In current master of LibreOffice the Plugin is working, so I think chances aren't that bad, that the plugin will work again in 7.5.5.

**Comment by ColeBantam** [comment]
One more update:
1.) The crash I mentioned before only happens with Snapshot builds of LanguageTool and are not due to an error in LibO 7.5.5.1. So, stable builds of LT-Plugin are working again in LibO 7.5.5.1.
2.) LT-Plugin version 6.1.1 seems to solve the performance issues! :) 👍 I will test that over the next days more intensive, but it looks very promising :)

**Comment by FredKruse** [comment]
I'm not at home. I will test the issue next week. The last stable should be LT 6.2.

**Comment by ColeBantam** [comment]
> I'm not at home. I will test the issue next week. The last stable should be LT 6.2.

There is no "LanguageTool-6.2.oxt" in Download-Folder, only a "LanguageTool-6.2.zip". And the Upload-Date of file "LanguageTool-stable.oxt" is 29-Dec-2022, so thats very unlikely to be Version 6.2 :D

**Comment by Tex2002ans** [comment]
> I checked this, and it seems that LibreOffice 7.5 breaks the LanguageTool-Plugin
>
> OK, this is an already known issue:
> 
> #8768
>
> [...]

Thanks for that link! Looks like it was an issue on LibreOffice 7.5's end.

LT not working will be fixed in LibreOffice 7.5.5.

I linked to the exact bug + gave a little more info here:

- https://github.com/languagetool-org/languagetool/issues/8768#issuecomment-1620830000

**Comment by FredKruse** [comment]
The issue is solved in the current versions

**Comment by ColeBantam** [comment]
I can confirm that the context-menu-slowdown is solved in 6.3. Thanks to all who helped squashing that bug :)
